tests/test_flatten.py

```diff
@@ -25,7 +28,11 @@ def teste_flatten_with_emptylist():
     assert flatten([]) == []
 
 
-# test the flatten function with a dictionary mocking it to assert _flatten_dict function is being called
+def teste_flatten_with_emptyDataFrame():
+    data = flatten(DataFrame())
+    assert data.equals(DataFrame())
```

Comment:
Created this test to assure if an empty `DataFrame` is passed aas `data`, it will be returned buy the early return.
Should I test if `is_empty` function is being called, right?
Just realized this now. I will update it soon...

Reply:
> Created this test to assure if an empty `DataFrame` is passed aas `data`, it will be returned buy the early return.

This test does not assure that. It ensures the result is the “same”, but not “when” it is returned. If you wanna make sure this is happening, you have to mock a method that happens _after_ the early return and [`.assert_not_called()`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_not_called).